### PR TITLE
Consume connections better in socket-level tests

### PR DESCRIPTION
**Pull request by hodbn**
Add a utility that allows better timing of test responses, instead of socket consuming.

In Windows, after the second operation on a remotely closed socket a `WSAECONNABORTED` is raised.
In our socket-level tests, we implement a dummy HTTP server that only consumes data from the client socket, therefore it can't know when to stop consuming and might close the socket too early. This might cause unnecessary `send/recv`s on the already closed socket, which in turn will raise `WSAECONNABORTED` when it could be avoided.

I've added the 2 tests that are currently skipped on Windows (since [#1951](https://github.com/urllib3/urllib3/pull/1951)).
This change should prevent `WSAECONNABORTED` when testing the aforementioned tests on Windows.

### CODE REVIEW

**Comment by pquentin** [comment]
@hodbn I merged master into your branch, since it was a prerequisite before merging this anyway

File: dummyserver/testcase.py
Comment (pquentin) [inline dummyserver/testcase.py:236]: Can you please move this import at the top of the file?

**Comment by pquentin** [comment]
This looks like black magic to me, but is great if it works! Would like to have another pair of eyes before I merge it

File: dummyserver/testcase.py
Comment (hodbn) [reply dummyserver/testcase.py]: Done :)

**Comment by hodbn** [comment]
> 
> 
> This looks like black magic to me, but is great if it works! Would like to have another pair of eyes before I merge it

I've added another sentence of docs, hopefully the class purpose is more clear now.

**Review by pquentin — APPROVED** [review]
Let's try that! Thanks.

**Comment by PleasantMachine9** [comment]
I'm guessing this is the change that makes the CI flaky?
https://github.com/urllib3/urllib3/pull/1970/checks?check_run_id=1136185374

At least the MacOS tests failed 2/2 times, and first it was the 3.8, then the 3.6 test, and different ones, with the only common thing being that they were both socket-related (Connection Refused).

**Comment by hodbn** [comment]
> I'm guessing this is the change that makes the CI flaky?
> https://github.com/urllib3/urllib3/pull/1970/checks?check_run_id=1136185374
> 
I don't think these failures are related to this change.
This change only affects `test_preserve_chunked_on_redirect` and `test_preserve_chunked_on_broken_connection` for now.

**Pull request merged**
